packages/gatsby-telemetry/src/ci.js

```diff
@@ -0,0 +1,60 @@
+const ci = require(`ci-info`)
+
+const CI_DEFINITIONS = [
+  getEnvFromCIInfo,
+  getEnvDetect({ key: `NOW_BUILDER_ANNOTATE`, name: `ZEIT Now` }),
+  getEnvDetect({ key: `NOW_REGION`, name: `ZEIT Now v1` }),
+  herokuDetect,
+  envFromCIandCIName,
+]
+
+function lookupCI() {
+  for (const fn of CI_DEFINITIONS) {
+    const res = fn()
+    if (res) {
+      return res
+    }
+  }
+  return null
+}
+const CIName = lookupCI()
+
+function isCI() {
+  return !!CIName
+}
+
+function getCIName() {
+  if (!isCI()) {
+    return null
+  }
+  return CIName
+}
+
+module.exports = { isCI, getCIName }
+
+function getEnvFromCIInfo() {
+  if (ci.isCI) return ci.name || `ci-info detected w/o name`
+  return null
+}
+
+function getEnvDetect({ key, name }) {
+  return function() {
+    if (process.env[key]) {
+      return name
+    }
+    return null
+  }
+}
+
+function herokuDetect() {
+  return /\.heroku\/node\/bin\/node/.test(process.env.NODE) && `Heroku`
```

Comment:
I'm not sure of reliability of this - I checked one of my Dynos and `process.env.NODE` is not set there:

```shell
~ $ node -e "console.log(process.env)"
{ _: '/app/.heroku/node/bin/node',
  NODE_HOME: '/app/.heroku/node',
  NODE_ENV: 'production',
  WEB_MEMORY: '512',
  DYNO: 'run.7506',
  PWD: '/app',
  HOME: '/app',
  PORT: '33175',
  MEMORY_AVAILABLE: '512',
  TERM: 'xterm-256color',
  WEB_CONCURRENCY: '1',
  SHLVL: '2',
  PATH:
   '/app/.heroku/node/bin:/app/.heroku/yarn/bin:/usr/local/bin:/usr/bin:/bin:/app/bin:/app/node_modules/.bin',
  PS1:
   '\\[\\033[01;34m\\]\\w\\[\\033[00m\\] \\[\\033[01;32m\\]$ \\[\\033[00m\\]' }
~ $ node -e "console.log(process.env.NODE)"
undefined
```

Reply:
I might completely missconfigure my dyno - but it also means people can run gatsby on Heroku in same way I did :P

Reply:
It appears to be set in the build process. But indeed they don't set any `heroku=1` like environment variables. And in my couple samples the build process has NODE set like this: `NODE=/tmp/build_8a43526a8849e690a3b67906d404e434/.heroku/node/bin/node`